plugins/woocommerce-blocks/assets/js/types/type-defs/currency.ts

```diff
@@ -33,14 +33,87 @@ export interface Currency {
 	thousandSeparator: string;
 }
 
+type CurrencyPrefix =
+	| 'kr' // Various Nordic currencies (e.g., Swedish Krona)
+	| '£' // British Pound Sterling
+	| '$' // US Dollar and others
+	| '€' // Euro (used as a prefix in most Western European countries)
+	| '؋' // Afghan Afghani
+	| '֏' // Armenian Dram
+	| 'Kz' // Angolan Kwanza
+	| '₼' // Azerbaijani Manat
+	| '৳' // Bangladeshi Taka
+	| 'Bs' // Venezuelan Bolívar
+	| 'P' // Botswana Pula
+	| '¥' // Japanese Yen and Chinese Yuan
+	| '₡' // Costa Rican Colón
+	| 'E£' // Egyptian Pound
+	| '₾' // Georgian Lari
+	| 'GH₵' // Ghanaian Cedi
+	| 'FG' // Guinean Franc
+	| 'Q' // Guatemalan Quetzal
+	| 'L' // Honduran Lempira and Albanian Lek
+	| 'Rp' // Indonesian Rupiah
+	| '₪' // Israeli Shekel
+	| '₹' // Indian Rupee
+	| '¥' // Japanese Yen
+	| '៛' // Cambodian Riel
+	| 'CF' // CFA Franc (Central and West Africa)
+	| '₩' // South and North Korean Won
+	| '₸' // Kazakhstani Tenge
+	| 'L£' // Lebanese Pound
+	| 'Ar' // Malagasy Ariary
+	| 'K' // Papua New Guinean Kina
+	| '₮' // Mongolian Tugrik
+	| 'RM' // Malaysian Ringgit
+	| '₦' // Nigerian Naira
+	| '₱' // Philippine Peso
+	| '₴' // Ukrainian Hryvnia
+	| '₫' // Vietnamese Dong
+	| 'R' // South African Rand
+	| 'ZK' // Zambian Kwacha
+	| 'Rs'; // Indian, Pakistani, and other rupees
+
+// Currencies typically used as a suffix (e.g., "50 Ft")
+type CurrencySuffix =
+	| 'KM' // Bosnian Convertible Mark
+	| 'Kč' // Czech Koruna
+	| 'Ft' // Hungarian Forint
+	| 'lei' // Romanian Leu
+	| '₽' // Russian Ruble
+	| 'RF' // Rwandan Franc
+	| 'Db' // São Tomé and Príncipe Dobra
+	| '฿' // Thai Baht
+	| 'zł' // Polish Zloty
+	| '₭'; // Lao Kip
+
+// Currencies that can appear as either prefix or suffix depending on region or context (e.g., "€50" or "50 €")
+type CurrencyBoth =
+	| '€' // Euro (prefix in most countries, suffix in some like Spain)
+	| '₫' // Vietnamese Dong
+	| '$' // Dollar in some Spanish-speaking countries
+	| '₡' // Costa Rican Colón
+	| '₩'; // Korean Won
 export interface CurrencyResponse {
 	currency_code: CurrencyCode;
-	currency_symbol: string;
+
+	currency_symbol:
+		| CurrencyPrefix
+		| CurrencySuffix
+		| CurrencyBoth
+		| ( string & {} ); // eslint-disable-line @typescript-eslint/ban-types
+
+	currency_prefix: CurrencyPrefix | CurrencyBoth;
```

Comment:
I guess it can be empty string the same as suffix 🤔 

```suggestion
	currency_prefix: '' | CurrencyPrefix | CurrencyBoth;
```